### PR TITLE
trivial: make permission failures for loading config non-fatal

### PR DESCRIPTION
**Pull request by superm1**
We already made the configuration files optional, so it's incongruent to make failures to read a non-essential file fatal.

This fixes running fwupdtool firmware-parse as non-root.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-config.c [hunk 1]
```diff
@@ -165,9 +165,16 @@ fu_config_reload(FuConfig *self, GError **error)
...
+			if (blob == NULL) {
+				if (g_error_matches(error_load, G_FILE_ERROR, G_FILE_ERROR_ACCES)) {
+					g_debug("%s", error_load->message);
+					continue;
+				}
+				g_propagate_error(error, g_steal_pointer(&error_load));
 				return FALSE;
+			}
 			fu_byte_array_append_bytes(buf, blob);
 		}
```
Comment (hughsie) [inline libfwupdplugin/fu-config.c:179]: maybe we drop the g_file_query_exists() and handle the extra failure case?

Comment (superm1) [reply libfwupdplugin/fu-config.c]: changed

File: libfwupdplugin/fu-config.c [hunk 2]
```diff
@@ -163,22 +163,29 @@ fu_config_reload(FuConfig *self, GError **error)
...
+		g_autoptr(GBytes) blob_item = NULL;
 		g_debug("trying to load config values from %s", item->filename);
-		if (g_file_query_exists(item->file, NULL)) {
-			g_autoptr(GBytes) blob = fu_bytes_get_contents(item->filename, error);
-			if (blob == NULL)
-				return FALSE;
-			fu_byte_array_append_bytes(buf, blob);
+		blob_item = fu_bytes_get_contents(item->filename, &error_load);
+		if (blob_item == NULL) {
+			if (g_error_matches(error_load, G_FILE_ERROR, G_FILE_ERROR_ACCES)) {
```
Comment (hughsie) [inline libfwupdplugin/fu-config.c:171]: is it EACCESS for not-found as well? my UNIX-foo is poor.

File: libfwupdplugin/fu-config.c [hunk 3]
```diff
@@ -163,22 +163,29 @@ fu_config_reload(FuConfig *self, GError **error)
...
+				g_debug("%s", error_load->message);
```
Comment (hughsie) [inline libfwupdplugin/fu-config.c:172]: maybe "ignoring config file: %s"

File: libfwupdplugin/fu-config.c (hunk 2 above)
Comment (superm1) [reply libfwupdplugin/fu-config.c:171]: no it should be `G_FILE_ERROR_EXIST``, added another case for it.

**Review by hughsie — APPROVED** [review]

**Pull request merged**